### PR TITLE
fix: TypeError: Cannot read properties of null (reading 'user_id')

### PR DESCRIPTION
**Pull request by iamsivin**
# Pull Request Template

## Description

This PR will fix an issue with the audit log page.

**Issue**
This PR addresses an issue where the page crashes and throws the error `TypeError: Cannot read properties of null (reading 'user_id')` when there is an update event in the audit log, and the updated agent is being deleted. This error occurs because the `auditable` object is `null`, leading to an attempt to access the `user_id` property of a null object

**Solution**
The fix ensures that the code properly handles cases where the `auditable` object is null before attempting to access the `user_id` property. Specifically, before setting the `translationPayload`, a check is added to verify if the auditable object is present. If auditable is present, it shows the `agent name`, otherwise, it returns a string `deleted user`. For eg:, "Shivam changed role, availability of deleted user to administrator, offline."

Fixes https://linear.app/chatwoot/issue/CW-3501/typeerror-cannot-read-properties-of-null-reading-user-id

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

**Steps:**

1. Enable audit logs for the account.
2. Then edit any other agent availability/role from agent's page, then save it.
3. Then go to the audit log page and see it's there. For eg:, "Shivam changed role,availability of Sivin to administrator,offline"
4. Go back to agent's page and delete the updated agent.
5. And go back to audit log page and see.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by scmmishra — DISMISSED** [review]

File: app/javascript/dashboard/helper/auditlogHelper.js [hunk 1]
```diff
@@ -89,11 +89,10 @@ function handleAccountUserCreate(auditLogItem, translationPayload, agentList) {
...
 function handleAccountUserUpdate(auditLogItem, translationPayload, agentList) {
-  if (auditLogItem.user_id !== auditLogItem.auditable.user_id) {
-    translationPayload.user = getAgentName(
-      auditLogItem.auditable.user_id,
-      agentList
-    );
+  if (auditLogItem.user_id !== auditLogItem.auditable?.user_id) {
+    translationPayload.user = auditLogItem.auditable
+      ? getAgentName(auditLogItem.auditable?.user_id, agentList)
+      : 'deleted user';
```
Comment (vishnu-narayanan) [inline app/javascript/dashboard/helper/auditlogHelper.js:95]: Can we use i18n here?
```suggestion
      :  this.$t(`AUDIT_LOGS.DELETED_USER`);
      
```

**Review by vishnu-narayanan — COMMENTED** [review]
Looks good. 

1. Can we use i18n to avoid hardcoding `deleted user`?
2. Could you add a test for this case in auditlog_helper spec?

File: app/javascript/dashboard/helper/auditlogHelper.js (hunk 1 above)
Comment (iamsivin) [reply app/javascript/dashboard/helper/auditlogHelper.js]: @vishnu-narayanan We can't do i18n in a helper file.

Comment (iamsivin) [reply app/javascript/dashboard/helper/auditlogHelper.js]: Done https://github.com/chatwoot/chatwoot/pull/9884/commits/711b0ce87a04127174a74e9306780bfe477fde59

**Review by vishnu-narayanan — APPROVED** [review]

**Pull request merged**